client/tests/integration/queries/asset_definition.rs

```diff
@@ -0,0 +1,33 @@
+#![allow(clippy::restriction)]
+
+use std::str::FromStr as _;
+
+use eyre::Result;
+use iroha_client::client;
+use iroha_data_model::prelude::*;
+use test_network::Peer as TestPeer;
+
+#[test]
+fn find_asset_definition_by_id() -> Result<()> {
+    let (_rt, _peer, test_client) = <TestPeer>::start_test_with_runtime();
+
+    let account_id: AccountId = "alice@wonderland".parse().expect("Valid");
+
+    // Registering a new asset definition
+    let asset_definition_id =
+        <AssetDefinition as Identifiable>::Id::from_str("test_coin#wonderland").expect("Valid");
+    let asset_definition = AssetDefinition::quantity(asset_definition_id.clone());
+    test_client.submit_blocking(RegisterBox::new(asset_definition.clone()))?;
+    test_client.submit_blocking(MintBox::new(
```

Comment:
Here is also worth querying and checking the result

```suggestion
    test_client.submit_blocking(RegisterBox::new(asset_definition.clone()))?;
    // Here
    test_client.submit_blocking(MintBox::new(
```